packages/plugin-truffle/src/utils/validations.ts

```diff
@@ -41,6 +47,21 @@ function reconstructSolcInputOutput(artifacts: TruffleArtifact[]): { input: Solc
       // We ignore it. If the contract is needed by the user it will fail later.
       continue;
     }
+    if (artifact.compiler.name !== 'solc') {
+      continue;
+    }
+
+    const versionMatch = artifact.compiler.version.match(/^([0-9.]+)\+commit/);
+    if (!versionMatch) {
+      throw new UpgradesError('Unknown or nightly compiler version');
+    }
+    const [artifactSolcVersion] = versionMatch;
```

Comment:
Currently this captures the whole match so it appears as `0.5.16+commit` in the manifest.  Was this intended to only capture the first group? 
```suggestion
    const artifactSolcVersion = versionMatch[1];
```